docs/coding_guidelines.md

```diff
@@ -0,0 +1,78 @@
+# Coding guidelines for OCS-CI project
+
+Lets follow the following guidelines for the Core Libraries and Test Scripts
+
+* Follow pep8 or use autopep8 to automagically conform to pep8 style.
+* Each Python Class is well documented for inheritence, methods and
+    attributes, Add sufficent comments for code block that is easy
+    to understand for others to review.
+* Code Readablity takes priority, Lets avoid crazy hacks and tricks 
+    and stick to straight forward code blocks that are easy to
+    understand for all.
+* For core libraries when possible please add the unittest to
+    minimize regression for many tests
+* Helper functions that are useful to wide variety of tests should
+    reside in utils and/or should be carefully reviewed to check if
+    they can actually be methods.
+* test_scripts will reside in test/ folders and will define run 
+    unction
+* test_scripts should return non zero exit status for failure and 0
+    for 'pass'
+
+**ex: test_scripts/test_ocs_operator.py**
+
+```python
+        def run(ocs_context, **kw):
+            # use ocs_context to deal with cluster
+            # run your test steps
+            if test_failed:
+                return False
+            else:
+                return Pass
+```
+
+* Do not use backslashes in theh code for line breaker!
+* Line length should be maximum 79 characters!
+* Try to follow this convention for brackets and indentions like in example below (This makes code much more readable):
+
+```python
+def function(
+    param1, param2, param3, param4
```

Comment:
Just testing comment, please change to longer name of parameters

Reply:
@vasukulkarni @RazTamir @clacroix12 here I see my comment, I can easily now see that fix is in commit which I see in the link bellow with [f69fc66](https://github.com/red-hat-storage/ocs-ci/pull/7/commits/f69fc66f48175885dce3a79d57daaf2da63f697f) commit. So I can see what was fixed and if it align with my comment. You can still click to [View changes](https://github.com/red-hat-storage/ocs-ci/pull/7/files/c011335c317357136f098039ba040c3ee59b476a) above and still see all the history.

Still I see possibility to merge this PR with Squash and merge with 5 commits:
<img width="976" alt="Screen Shot 2019-04-19 at 1 25 51 AM" src="https://user-images.githubusercontent.com/16169617/56397042-29ca5400-6242-11e9-9d04-7629b3b920d8.png">

So are you agree with this approach I am suggesting?

Reply:
The thing about squash and merge is, it can squash all things to one commit, but if we have 2 different commits, a, a', b, b` , I dont think it can squash a' on a and b' and b(shylesh's pr), users still have to handle it, still its a useful feature to have.